crates/trie/db/src/proof.rs

```diff
@@ -11,145 +11,142 @@ use reth_trie::{
+        &self,
         input: TrieInput,
         address: Address,
         slots: &[B256],
     ) -> Result<AccountProof, StateProofError> {
         let nodes_sorted = input.nodes.into_sorted();
         let state_sorted = input.state.into_sorted();
-        Self::from_tx(tx)
+        self.clone()
             .with_trie_cursor_factory(InMemoryTrieCursorFactory::new(
-                DatabaseTrieCursorFactory::new(tx),
+                self.trie_cursor_factory().clone(),
                 &nodes_sorted,
             ))
             .with_hashed_cursor_factory(HashedPostStateCursorFactory::new(
-                DatabaseHashedCursorFactory::new(tx),
+                self.hashed_cursor_factory().clone(),
                 &state_sorted,
             ))
             .with_prefix_sets_mut(input.prefix_sets)
             .account_proof(address, slots)
     }
 
     fn overlay_multiproof(
-        tx: &'a TX,
+        &self,
         input: TrieInput,
         targets: MultiProofTargets,
     ) -> Result<MultiProof, StateProofError> {
         let nodes_sorted = input.nodes.into_sorted();
         let state_sorted = input.state.into_sorted();
-        Self::from_tx(tx)
+        self.clone()
             .with_trie_cursor_factory(InMemoryTrieCursorFactory::new(
-                DatabaseTrieCursorFactory::new(tx),
+                self.trie_cursor_factory().clone(),
                 &nodes_sorted,
             ))
             .with_hashed_cursor_factory(HashedPostStateCursorFactory::new(
-                DatabaseHashedCursorFactory::new(tx),
+                self.hashed_cursor_factory().clone(),
                 &state_sorted,
             ))
             .with_prefix_sets_mut(input.prefix_sets)
             .multiproof(targets)
     }
 }
 
-/// Extends [`StorageProof`] with operations specific for working with a database transaction.
-pub trait DatabaseStorageProof<'a, TX> {
-    /// Create a new [`StorageProof`] from database transaction and account address.
-    fn from_tx(tx: &'a TX, address: Address) -> Self;
+// Add constructor for Proof
+impl<'a, TX: DbTx> Proof<DatabaseTrieCursorFactory<'a, TX>, DatabaseHashedCursorFactory<'a, TX>> {
+    /// Create a new [Proof] instance from database transaction.
+    pub fn from_tx(tx: &'a TX) -> Self {
+        Self::new(DatabaseTrieCursorFactory::new(tx), DatabaseHashedCursorFactory::new(tx))
+    }
+}
 
+/// Extends [`StorageProof`] with operations specific for working with a database transaction.
+pub trait DatabaseStorageProof {
     /// Generates the storage proof for target slot based on [`TrieInput`].
     fn overlay_storage_proof(
-        tx: &'a TX,
-        address: Address,
+        &self,
         slot: B256,
         storage: HashedStorage,
     ) -> Result<reth_trie::StorageProof, StateProofError>;
 
     /// Generates the storage multiproof for target slots based on [`TrieInput`].
     fn overlay_storage_multiproof(
-        tx: &'a TX,
-        address: Address,
+        &self,
```

Comment:
we need to keep `address` in the trait I think? Even if we already have the address in the `DatabaseStorageProof` struct, we should keep the behavior consistent with how we did it previously, which was hashing the input addr and returning the storage proof for that hashed addr